### DISCUSSION TITLE
Dynamically changing cli prompt

### DISCUSSION THREAD

**Discussion by stephb9959** [daniele77/cli#198] [Q&A]
Hi, 

great project. I just found this and I love it.

We would like to do something like the following:

```
cli> device device_name_1
cli device_name_1>  show
...
```
I have reviewed the examples and the code and I cannot see an obvious way to do this.  Do I need better glasses or this is not available?

Thank you!

**Comment by daniele77** [comment]
Thank you for using CLI library, @stephb9959 !
Currently, it's impossible to change the prompt programmatically: it reports the name of the current menu/submenu.
So, when you are in the menu "foo" and you change it to "bar" you got:

    foo>                <-- you're in the submenu "foo"
    foo> bar            <-- you enter the submenu "bar" using the "bar" command
    bar>                <-- you're in the submenu "bar", now

However, if you think it's an important feature, you can fill a new GitHub issue with the request to add a function to change the current prompt shown by the library.

(If you like the library, please consider becoming a sponsor by clicking the heart button at the top of this page. You can also donate once.)

**Reply by 10669675** [reply]
I also appreciate this feature going live. Could you please let me know if there is a release plan for it?

**Reply by daniele77** [reply]
This feature requires #49 first because a user should be able to change the prompt only on its session. Unfortunately, #49 requires breaking the APIs (the command callbacks would take a `UserSession` object parameter instead of the current `std::ostream`) and this scare me a bit :-)

**Comment by stephb9959** [comment]
Thanks, @daniele77 !

I am trying to duplicate the behavior you see in Cisco equipment CLI for example. You can enter a menu level and then by entering a device's name, MAC address, or some identifier, the prompt changes to the 'menu name' + 'identifier'. The user now knows that they are dealing with a specific device. The identifier is unpredictable, so you cannot create a menu of that name when you set up the CLI structure. I'll file an issue.

**Comment by daniele77** [comment]
Fine, it seems a nice feature to have in the library.
We'll try to develop it soon.
Thank you very much for the suggestion and for using our library.
